src/Board.cpp

```diff
@@ -0,0 +1,168 @@
+#include "Board.h"
+
+Board::Board(const std::string& board)
+	: m_playerIsWhite(true)
+{
+    Factory factory;
+    for (auto i = 0; i < board.size(); ++i)
+    {
+        auto row = i / ROWS;
+        auto col = i % COLS;
+        if (board[i] == 'K')
+            m_kingLoc[1] = { row, col };
+
+        else if (board[i] == 'k')
+            m_kingLoc[0] = { row, col };
+
+        m_board[row][col] = factory.createPiece(board[i]);
+    }
+}
+
+int Board::getMove(const std::string& move)
+{
+    Location src = { move[0] - 'a', move[1] - '1' };
+    Location dst = { move[2] - 'a', move[3] - '1' };
+
+    if (isEmpty(src))
+        return 11;
+    if (isRival(src))
+        return 12;
+    if (isCurrentPiecePlayer(dst))
+        return 13;
+    if (!m_board[src.x][src.y]->isCorrectMove(src, dst) || isWayBlocked(src, dst))
+        return 21;
+    if (moveCausesChess(src, dst))
+        return 31;
+    update(src, dst);
+    if (mate())
+        return 41;
+
+    return 42;
+}
+
+
+
+void Board::initBoard()
+{
+    for (size_t row = 0; row < ROWS; ++row)
+        for (size_t col = 0; col < COLS; ++col)
+            m_board[row][col] = nullptr;

```

Comment:
Instead of manually setting each element of m_board to nullptr using a nested for loop, you can use std::fill or std::fill_n to initialize the entire m_board with nullptr in a more concise way. like this: std::fill_n(&m_board[0][0], ROWS * COLS, nullptr);